Form/Type/DateRangeType.php

```diff
@@ -20,15 +20,28 @@
 class DateRangeType extends AbstractType
 {
     /**
-     * @var TranslatorInterface
+     * @var TranslatorInterface|null
+     *
+     * @deprecated translator property is deprecated since version 3.x, to be removed in 4.0
      */
     protected $translator;
 
     /**
-     * @param TranslatorInterface $translator
+     * @param TranslatorInterface|null $translator
+     *
+     * @deprecated translator dependency is deprecated since version 3.x, to be removed in 4.0
      */
-    public function __construct(TranslatorInterface $translator)
+    public function __construct(TranslatorInterface $translator = null)
     {
+        // check if class is overloaded and notifiy about removing deprecated translator
+        if ($translator !== null && get_class($this) !== get_class() && get_class($this) !== 'Sonata\CoreBundle\Form\Type\DateRangePickerType') {
```

Comment:
@gregoire This is a way how to notify users, that has overloaded our native for types.